lib/escape.c

```diff
@@ -132,6 +144,28 @@ char *curl_easy_escape(struct Curl_easy *data, const char *string,
   return ns;
 }
 
+char *curl_easy_escape(struct Curl_easy *data, const char *string,
+                       int inlength)
+{
+  return curl_easy_escape_flexible(data, string, inlength, &Curl_isunreserved);
+}
+
+/* escape form data per RFC1866 */
+char *curl_easy_escape_form(struct Curl_easy *data, const char *string,
+                            int inlength)
+{
+  char *enc = curl_easy_escape_flexible(data, string, inlength,
+                                        &form_isunreserved);
+
+  /* replace space with + */
+  for(int enc_index = 0; enc_index < strlen(enc); enc_index++) {
```

Comment:
variable declarations within the `for()` is not C89 compliant.

I think the strlen() call should rather be done once outside of the loop than for every lap.